lib/rdf/util/file.rb

```diff
@@ -233,6 +233,54 @@ def self.open_url base_uri, options
       end
     end
 
+    ##
+    # Use Hurley for retrieving resources
+    # @since 1.2
+    class HurleyAdapter < HttpAdapter
+      class <<self
+        ##
+        # Set the Hurley::Client to use for retrieving RDF resources
+        def client= client
+          @client = client
+        end
+
+        ##
+        # Get the Hurley::Client to use for retrieving RDF resources,
+        # or a default connect that follows redirects.
+        def client
+          @client ||= Hurley::Client.new
+        end
+      end
+
+      # @see HttpAdapter.open_url
+      # @param [String] base_uri to open
+      # @param  [Hash{Symbol => Object}] options
+      # @return [RemoteDocument, Object] A {RemoteDocument}.
+      def self.open_url base_uri, options
+        response = client.get(base_uri) do |req|
+          headers(options).each do |k,v|
+            req.header[k] = v
+          end
+        end
+
+        if response.success?
+          # found object
+
+          # If a Location is returned, it defines the base resource for this file, not it's actual ending location
+          document_options = {
+            base_uri:     RDF::URI((response.location || response.request).url.to_s),
+            charset:      Encoding::UTF_8,
+            code:         response.status_code,
+            headers:      response.header
```

Comment:
Right now, this leaks the [`Hurley::Header`](https://github.com/lostisland/hurley/blob/master/lib/hurley/header.rb) object into RemoteDocument. They have some nice logic about header canonicalization, but it may be better to just map it back into the snake-cased symbols that `rdf` expects..